libs/oidc/common/src/lib/utils/email/mailer.util.ts

```diff
@@ -3,18 +3,59 @@ import Mail from 'nodemailer/lib/mailer';
 
 import { User, UserPasswordReset } from '@tamu-gisc/oidc/common';
 
+export type NodeMailerServices = 'ethereal' | 'gmail' | 'tamu-relay';
 export class Mailer {
-  private static user = 'kaitlyn.schimmel@ethereal.email';
-  private static password = 'Pe6D9DhkgUDyqyBMeg';
-  private static transporter: Mail = nodemailer.createTransport({
-    host: 'smtp.ethereal.email',
-    port: 587,
-    secure: false,
-    auth: {
-      user: Mailer.user,
-      pass: Mailer.password
+  private static transporter: Mail;
+  private static service: NodeMailerServices;
+
+  // https://dev.to/chandrapantachhetri/sending-emails-securely-using-node-js-nodemailer-smtp-gmail-and-oauth2-g3a
+  // 2LO https://nodemailer.com/smtp/oauth2/#oauth-2lo
+  public static build(
+    service: NodeMailerServices,
+    config?: {
+      user: string;
+      accessToken: string;
+      clientId: string;
+      clientSecret: string;
+      refreshToken: string;
+    }
+  ) {
+    Mailer.service = service;
+
+    switch (service) {
+      case 'ethereal':
+        // use ethereal email and print out links to console (debugging / dev)
+        Mailer.transporter = nodemailer.createTransport({
+          host: 'smtp.ethereal.email',
+          port: 587,
+          secure: false,
+          auth: {
+            user: 'kaitlyn.schimmel@ethereal.email',
+            pass: 'Pe6D9DhkgUDyqyBMeg'
+          }
```

Comment:
I can only assume this shouldn't be here?

Reply:
That depends. It's a free account that was automatically generated. Should it be there? No. Does it really matter? Not particularly.

Reply:
![image](https://user-images.githubusercontent.com/3969818/126570572-bce9d680-5249-4d9d-82e9-fc1a2dc3fbeb.png)